common/src/hand_off.rs

```diff
@@ -120,9 +121,21 @@ impl Default for FirmwareHandoffTable {
 
 impl FirmwareHandoffTable {
     /// Perform valdity check of the table's data.
+    /// The fields below should have been populated by ROM with
+    /// valid data before it transfers control to mutable code.
     pub fn is_valid(&self) -> bool {
-        self.fht_marker == FHT_MARKER && self.fmc_cdi_kv_idx != FHT_INVALID_IDX
+        self.fht_marker == FHT_MARKER
+            && self.fmc_cdi_kv_idx != FHT_INVALID_IDX
+            && self.manifest_load_addr != FHT_INVALID_ADDRESS
+            && self.fmc_pub_key_x_dv_idx != FHT_INVALID_IDX
+            && self.fmc_pub_key_y_dv_idx != FHT_INVALID_IDX
+            && self.fmc_cert_sig_r_dv_idx != FHT_INVALID_IDX
+            && self.fmc_cert_sig_s_dv_idx != FHT_INVALID_IDX
+            && self.rt_tci_dv_idx != FHT_INVALID_IDX
+            && self.rt_fw_entry_point != FHT_INVALID_ADDRESS
```

Comment:
please add additional checks....

1. self.rt_fw_load_addr != FHT_INVALID_ADDRESS
2. self.fips_fw_load_addr == FHT_INVALID_ADDRESS (since we don't yet support anything else)
3. anything else that ROM is required to fill per the FMC spec

Reply:
Should also make sure the RT entry point is inside the locked ICCM memory, and that attacker is not somehow trying to get us to jump to an address in DCCM

Reply:
As per discussion with @FerralCoder  , we will add more extensive checks in the FMC and RT  and keep the scope of is_valid as light as possible , checking if fields have been populated by ROM to avoid feature creep inside the common crate.